adapters/grpc/document_grpc.go

```diff
@@ -110,7 +124,27 @@ func NewDocumentServer(docPlugin sdk.DocumentService) pb.DocumentServiceServer {
 	}
 }
 
-func toPbDoc(doc *sdk.Document) (*pb.Document, error) {
+// isInvalidSubCollectionDepth - returns an error if the provided collection exceeds the maximum supported
```

Comment:
Good addition. Currently most of the validation parameter validation code is in the file `/plugins/documents/document.go`.  This is used across all the document services.  I think the parameter validation code should be centralized.  

Next question is whether we perform validation in the adaptor layer.  The advantages of this is we protect against the plugin implementations which forget to do this. However is is probably only good for relatively simple validations, like what we currently do.  

Having the plugin services also enforce their contracts isn't terrible the services self contained and we ensure their behavior is consistent with the service tests.

Currently there there is the possibility of a NPE with the collection parameter, being null.

Reply:
Also good if we develop a pattern for across all our services. Nice error message by the way :)

Reply:
Moved the code as suggested, agree it's better to keep it together and it would be worth developing a pattern.